services/QuillLMS/app/services/classlink_integration/classroom_persister.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module ClasslinkIntegration
+  class ClassroomPersister < ApplicationService
+    attr_reader :incoming_classroom, :existing_provider_classrooms, :classlink_instance_id
+
+    def initialize(incoming_classroom:, existing_provider_classrooms:, classlink_instance_id:)
+      @incoming_classroom = incoming_classroom
+      @existing_provider_classrooms = existing_provider_classrooms
+      @classlink_instance_id = classlink_instance_id
+    end
+
+    def run
+      existing_classroom = existing_provider_classrooms.find do |ec|
```

Comment:
Consider making extracting this as a private method to simplify this long method.
```ruby
private def existing_classrooms
  @existing_classrooms || = existing_provider_classrooms.find do |ec|
    ec.external_id == incoming_classroom[:sourced_id]
  end
end